lib/pure/asyncfutures.nim

```diff
@@ -1,8 +1,18 @@
+include "system/inclrtl"
+
+import os, tables, strutils, times, heapqueue, options, deques
 
 # TODO: This shouldn't need to be included, but should ideally be exported.
 type
+  CallbackFunc = proc () {.closure, gcsafe.}
+
+  CallbackList = object
```

Comment:
I don't think it's possible - the first item needs to be inlined into `FutureVar` (to save alloc in common case).